### PR TITLE
Intra account impersonation enable/disable functionality added

### PR DESCRIPTION
**Pull request by nkarunarathne**

### CODE REVIEW

File: src/modules/cb_user_auth.erl [hunk 1]
```diff
@@ -139,9 +139,22 @@ authorize_nouns(_, _Nouns, _) -> 'false'.
...
-    lager:info("auth token's account id matches request's account id ~s, not allowed", [RequestAccountId]),
-    forbidden(Context);
+maybe_authorize_impersonation(Context, RequestAccountId, RequestUserId, RequestAccountId) ->
+    {'ok', RootAccountId} = kapps_util:get_master_account_id(),
+    AuthToken = cb_context:auth_token(Context),
+    {'ok', AuthJObj} = kz_auth:validate_token(AuthToken),
+    Method = kz_json:get_string_value(<<"method">>, AuthJObj),
+    case
+        (RootAccountId == RequestAccountId)
+        andalso (Method == "cb_api_auth")
```
Comment (jamesaimonetti) [inline src/modules/cb_user_auth.erl:149]: Let's prefer binaries over strings unless required to use strings. `Method = kz_json:get_ne_binary_value(<<"method">>, AuthJObj)` should be preferred, as binary comparison is more efficient that list comparison.

Comment (jamesaimonetti) [reply src/modules/cb_user_auth.erl:149]: Also, it is preferred to use `=:=` and `=/=` for equality checks.

File: src/modules/cb_user_auth.erl [hunk 2]
```diff
@@ -139,9 +139,22 @@ authorize_nouns(_, _Nouns, _) -> 'false'.
...
+    of
+        false ->
```
Comment (jamesaimonetti) [inline src/modules/cb_user_auth.erl:151]: `'false'` and `'true'` - ticky the atoms :) you can run `make elvis` from the kazoo root dir to help highlight these.

**Review by jamesaimonetti — CHANGES_REQUESTED** [review]

File: src/modules/cb_user_auth.erl [hunk 3]
```diff
@@ -139,9 +139,22 @@ authorize_nouns(_, _Nouns, _) -> 'false'.
 maybe_authorize_impersonation(Context, _RequestAccountId, _RequestUserId, 'undefined') ->
     lager:info("no auth account on auth token"),
     forbidden(Context);
-maybe_authorize_impersonation(Context, RequestAccountId, _RequestUserId, RequestAccountId) ->
-    lager:info("auth token's account id matches request's account id ~s, not allowed", [RequestAccountId]),
-    forbidden(Context);
+maybe_authorize_impersonation(Context, RequestAccountId, RequestUserId, RequestAccountId) ->
```
Comment (icehess) [inline src/modules/cb_user_auth.erl:142]: If the the intend is to allow only `cb_api_auth` when the account ids are same, then change it to something like this:

```
    lager:info("auth token's account id matches request's account id ~s, checking if auth method was cb_api_token", [RequestAccountId]),
    case cb_context:is_superduper_admin(Context) of
        'false' ->
            lager:info("auth account id ~s is not superduper", [AuthAccountId]),
            forbidden(Context);
        'true' ->
            maybe_authorize_impersonation_via_api_token(Context, RequestAccountId, RequestUserId, AuthAccountId
                                                           ,kz_json:get_value(<<"method">>, cb_context:auth_doc(Context))
                                                           );
    end;
```

File: src/modules/cb_user_auth.erl [hunk 4]
```diff
@@ -139,9 +139,22 @@ authorize_nouns(_, _Nouns, _) -> 'false'.
...
+    {'ok', RootAccountId} = kapps_util:get_master_account_id(),
+    AuthToken = cb_context:auth_token(Context),
+    {'ok', AuthJObj} = kz_auth:validate_token(AuthToken),
```
Comment (icehess) [inline src/modules/cb_user_auth.erl:145]: fyi We already validate the token! you just to `kz_json:get_string_value(<<"method">>, cb_context:auth_doc(Context))`,

But anyway remove this and use the code from comment above.

**Review by icehess — CHANGES_REQUESTED** [review]

**Pull request closed without merging**